apps/src/templates/teacherDashboard/ChangeShareSettingDialog.jsx

```diff
@@ -0,0 +1,97 @@
+import React, {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import i18n from '@cdo/locale';
+import {Heading1} from '../../lib/ui/Headings';
+import BaseDialog from '../BaseDialog';
+import Button from '../Button';
+import {sectionShape} from './shapes';
+import DialogFooter from "./DialogFooter";
+import PadAndCenter from './PadAndCenter';
+import {isSaveInProgress, updateShareSetting} from './teacherSectionsRedux';
+
+class ChangeShareSettingDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    handleClose: PropTypes.func.isRequired,
+    onSharingChanged: PropTypes.func.isRequired,
+    sectionId: PropTypes.number.isRequired,
+    // Provided by Redux
+    section: sectionShape,
+    isSaveInProgress: PropTypes.bool,
+    updateShareSetting: PropTypes.func.isRequired,
+    disableSharing: PropTypes.bool.isRequired
+  };
+
+  componentWillReceiveProps = (nextProps) => {
+    if (this.props.isSaveInProgress && !nextProps.isSaveInProgress) {
+      this.props.onSharingChanged();
+    }
+  };
```

Comment:
for life cycle methods like this, you dont need to do this pattern of creating a per instance method, and should instead just do `componentWillReceiveProps(nextProps) {`